lib/presentation/widgets/m3_widgets/buttons/filter_list_segmented_button.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter/material.dart';
+import 'package:trip_tally/presentation/theme/app_colors.dart';
+import 'package:trip_tally/presentation/theme/app_dimensions.dart';
+import 'package:trip_tally/presentation/theme/app_paths.dart';
+import 'package:trip_tally/presentation/utils/enums/context_extensions.dart';
+import 'package:trip_tally/presentation/utils/enums/transport_type.dart';
+import 'package:trip_tally/presentation/widgets/m3_widgets/cards/expandable_info_card.dart';
+import 'package:trip_tally/presentation/widgets/m3_widgets/cards/list_of_elevated_info_cards.dart';
+
+class FilterListSegmentedButton extends StatefulWidget {
+  const FilterListSegmentedButton({super.key});
+
+  @override
+  State<FilterListSegmentedButton> createState() => _FilterListSegmentedButtonState();
+}
+
+class _FilterListSegmentedButtonState extends State<FilterListSegmentedButton> {
+  List<bool> isSelected = [true, false, false];
+  List<String> items = ['banana', 'apple', 'pomegranate'];
+  List<String> filteredItems = [];
+
+  @override
+  void initState() {
+    super.initState();
+    filterList();
+  }
+
+  void filterList() {
+    setState(
+      () {
+        if (isSelected[0]) {
+          filteredItems = items.where((item) => item.startsWith('A')).toList();
+        } else if (isSelected[1]) {
+          filteredItems = items.where((item) => item.startsWith('B')).toList();
+        } else if (isSelected[2]) {
+          filteredItems = items;
+        }
+      },
+    );
+  }
```

Comment:
I think you can remove the filtering logic for now as it will be done in the backend. By tapping on the filter category you will send a request to backend to return the correct list